cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1573,25 +1576,50 @@ pub(crate) fn emit(
             emit_std_reg_mem(sink, prefix, opcode, 2, *src, dst, RexFlags::clear_w());
         }
 
-        Inst::XmmToGpr { op, src, dst } => {
-            let (rex, prefix, opcode) = match op {
-                SseOpcode::Movd => (RexFlags::clear_w(), LegacyPrefix::_66, 0x0F7E),
-                SseOpcode::Movq => (RexFlags::set_w(), LegacyPrefix::_66, 0x0F7E),
+        Inst::XmmToGpr {
+            op,
+            src,
+            dst,
+            dst_size,
+        } => {
+            let (prefix, opcode, dst_first) = match op {
+                SseOpcode::Movd => (LegacyPrefix::_66, 0x0F7E, false),
```

Comment:
nit: could you add a comment here, explaining how the Movd/Movq cases are distinguished?  Presumably via REX.W ?  Because the two match- arms here are identical, and it's confusing to look at at first (kinda looks like a bug :)

Reply:
Ah, good catch! At first, when the cvt instructions were not handled here, this would also set the RexFlags. Now we could unify movd and movq into a single enum variant, since they effectively have the same effect here. Will consider for later; it's nice to have the proper name in the enum list...